calblueprint/ogsc
# Add DashboardLayout component, AuthContext wrapper for app

**Authentication:**
* Creates /users/me API route to fetch the currently authenticated user
* Adds AuthContext provider at the top _app level for all routes prefixed with a "user role type", i.e. `/admin`, `/player`, `/donor`, `/mentor`. Consumers can access the AuthContext with a new hook, `useSessionInfo`, that performs the correct typecast. App redirects automatically to login for unauthenticated routes.

**Admin dashboard components/styling:**
* Adds DashboardLayout component to wrap around all views that should take on the left-sidebar user dashboard layout (when authenticated)
* Updates Tailwind config with wireframing colors (these are temporary. A key is attached below for referencing the colors defined in Tailwind to the ones in the Figma.)
* Refactors Icon component to consolidate all SVGs in one place (+ allow for them to be resized and recolored by stripping off fill/stroke properties)
* Creates standard Button component
* Moves current admin dashboard into `/admin/users` page, adds new `/admin/invite` page

## PR Checklist

Does your branch (check if true):


## Screenshots

Tailwind new color mappings (temporary wireframing colors):

<img src="https://user-images.githubusercontent.com/2977329/97121818-4fa9a200-16de-11eb-8e48-4e21f1b67969.png" width="500">


![image](https://user-images.githubusercontent.com/2977329/97120441-3b60a780-16d4-11eb-8d1f-33d6fd1e6515.png)

## Test Plan
* Visit `/admin/invite`. Verify that it redirects you to the login page.
* Sign in with `admin@ogsc.dev:password` (seeded, depends on #21).
* Visit `/admin/invite`. Observe that it displays a button with a plus icon (non-functional).
* Use the navbar to click on "All Users". Observe that it loads the page for `/admin/users`.

CC: @erinysong
